### PR TITLE
WebKit export of https://bugs.webkit.org/show_bug.cgi?id=188189

### PR DESCRIPTION
**Pull request by darsh7807**
WebKit export from bug: [Perform a microtask checkpoint before creating a custom element](https://bugs.webkit.org/show_bug.cgi?id=188189)

### CODE REVIEW

File: custom-elements/perform-microtask-checkpoint-before-construction.html [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+<head>
+<title>Custom Elements: create an element for a token must perform a microtask checkpoint</title>
+<meta name="author" title="Ryosuke Niwa" href="mailto:rniwa@webkit.org">
+<meta name="assert" content="When the HTML parser creates an element for a token, it must perform a microtask checkpoint before invoking the constructor">
+<meta name="help" content="https://html.spec.whatwg.org/multipage/parsing.html#create-an-element-for-the-token">
+<meta name="help" content="https://html.spec.whatwg.org/multipage/webappapis.html#perform-a-microtask-checkpoint">
+<meta name="help" content="https://html.spec.whatwg.org/multipage/parsing.html#adoption-agency-algorithm">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/custom-elements-helpers.js"></script>
```
Comment (fred-wang) [inline custom-elements/perform-microtask-checkpoint-before-construction.html]: mmh, it seems you ignored my comment https://github.com/web-platform-tests/wpt/pull/12447#discussion_r232938117 on the other PR. I thought "/resources/custom-elements-helper.js" gets the element at the root, so I think it should be "resources/custom-elements-helper.js" instead (without the initial /). Not sure why this is still working...

cc @Ms2ger

**Review by fred-wang — COMMENTED** [review]
Tests pass on Firefox but the second test fails on Chromium. Maybe @domenic can confirm this is expected and will be fixed in future Chromium release.

https://travis-ci.org/web-platform-tests/wpt/jobs/459042902
https://travis-ci.org/web-platform-tests/wpt/jobs/459042903#L625

Have you tried again with a trunk build of WebKit, just to be sure?

File: custom-elements/perform-microtask-checkpoint-before-construction.html (hunk 1 above)
Comment (darsh7807) [reply custom-elements/perform-microtask-checkpoint-before-construction.html]: @fred-wang  Just checked this  test is passing on webkit gtk

Comment (fred-wang) [reply custom-elements/perform-microtask-checkpoint-before-construction.html]: OK, but can you please fix the links to custom-elements-helper.js here and the one added in https://github.com/web-platform-tests/wpt/pull/12447 ?

**Comment by domenic** [comment]
@tkent-google or @hayatoito can you check that this failure is expected and will be fixable in Chrome?

File: custom-elements/perform-microtask-checkpoint-before-construction-expected.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+PASS HTML parser must perform a microtask checkpoint before constructing a custom element
+PASS HTML parser must perform a microtask checkpoint before constructing a custom element during the adoption agency algorithm
```
Comment (rniwa) [inline custom-elements/perform-microtask-checkpoint-before-construction-expected.txt]: We shouldn't be adding `-expected` files to WPT. These are WebKit specific files meant to keep track of the current test result.

**Comment by rniwa** [comment]
This is an importation of a test from https://trac.webkit.org/changeset/234944. In the future, please mention that in the commit even if you kept the author information in the file itself; a proper attribution is essential.

**Comment by fred-wang** [comment]
> This is an importation of a test from https://trac.webkit.org/changeset/234944. In the future, please mention that in the commit even if you kept the author information in the file itself. Because I contribute to WebKit on behalf of Apple, the author of the file and therefore the entity who hols the copyright to the test is Apple. Without a proper hyperlink in the commit message, discovering that attribution is hard unless you know who I am.
> 
> Most of WPT and browser vendor folks probably know or recognize my name one way or another, and can probably infer things from my `@webkit.org` email address but that won't hold true in 20-30 years from now.

Right, I edited the issue title and description to match the behavior of Youenn's export script. We should ensure we have this reference when we squash/merge this.

File: custom-elements/perform-microtask-checkpoint-before-construction.html (hunk 1 above)
Comment (fred-wang) [reply custom-elements/perform-microtask-checkpoint-before-construction.html]: OK, now I understand how the bad path was "fixed"... :-/

@darsh7807 Next time, please reply to questions we ask during review requests and be more explicit about how you addressed them. The move you silently did in https://github.com/web-platform-tests/wpt/pull/12447/commits/e342aed77e168325e86db3577c949c41b7f18f70 was not obvious from the commit message. Thank you.

Comment (darsh7807) [reply custom-elements/perform-microtask-checkpoint-before-construction.html]: Hi, sorry for the delay in reply I had build issues on webkit, I have now corrected the file path now,

File: custom-elements/perform-microtask-checkpoint-before-construction-expected.txt (hunk 2 above)
Comment (darsh7807) [reply custom-elements/perform-microtask-checkpoint-before-construction-expected.txt]: Thanks, I have removed it now!

**Comment by rniwa** [comment]
I've already moved `custom-elements-helpers.js` in https://github.com/web-platform-tests/wpt/pull/14295 so you'd have to revert that change.

**Comment by darsh7807** [comment]
Created new pull request, https://github.com/web-platform-tests/wpt/pull/14314 
I messed up this one!

**Pull request closed without merging**